dbcon/joblist/tuple-bps.cpp

```diff
@@ -1883,196 +1984,282 @@ void TupleBPS::sendPrimitiveMessages()
     tplLock.unlock();
 }
 
-struct _CPInfo
-{
-    _CPInfo(int64_t MIN, int64_t MAX, uint64_t l, bool val) : min(MIN), max(MAX), LBID(l), valid(val) { };
-    _CPInfo(int128_t BIGMIN, int128_t BIGMAX, uint64_t l, bool val) : bigMin(BIGMIN), bigMax(BIGMAX), LBID(l), valid(val) { };
-    union
-    {
-        int128_t bigMin;
-        int64_t min;
-    };
-    union
-    {
-        int128_t bigMax;
-        int64_t max;
-    };
-    uint64_t LBID;
-    bool valid;
-};
-
-void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
+void TupleBPS::processByteStreamVector(vector<boost::shared_ptr<messageqcpp::ByteStream>>& bsv,
+                                       uint32_t begin, uint32_t end, vector<_CPInfo>& cpv, RowGroupDL* dlp,
```

Comment:
I would suggest to make these variables constant if the function semantics allows this.